test/framework/src/main/java/org/elasticsearch/index/mapper/MappingLookupUtils.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.index.analysis.NamedAnalyzer;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class MappingLookupUtils {
+    public static MappingLookup fromTypes(MappedFieldType... types) {
+        return fromTypes(Arrays.asList(types));
+    }
+
+    public static MappingLookup fromTypes(List<MappedFieldType> fields) {
+        List<FieldMapper> mappers = new ArrayList<>();
+        List<RuntimeFieldType> runtimeFields = new ArrayList<>();
+        for (MappedFieldType type : fields) {
+            if (type instanceof RuntimeFieldType) {
+                runtimeFields.add((RuntimeFieldType) type);
+            } else {
+                mappers.add(mockFieldMapper(type));
+            }
```

Comment:
not a fan of this `if` either. I would rather do one of the following two things:

1) subclass MappingLookup and override only the needed methods, that way we don't have to create a field mapper at all. not sure if till work everywhere, but it's what were previously doing in QueryShardContextTests.

2) have a different method that accepts RuntimeFieldTypes. Along these lines I was wondering if the following method needs to be public, and if the analyzers logic is needed in all the callers or should be rather be pushed to the callers that need it, in which case we could make callers provide field mappers instead of mapped field types as they'd only need to do  `new MockFieldMapper(fieldType)`.

Reply:
I think it'd be pretty fair to move the analyzers change into `MockFieldMapper`. It is *weird* if the analyzers don't line up with the `MappedFieldType`'s analyzers. Sort of a "testing is weird" thing.

I don't much like the subclassing idea because `MappedFieldType` does a lot of stuff when we build it and I kind of think we should let it keep doing it. I'd be all for simplifying that stuff, I guess, but as it stands now I like using it raw. I'm quite happy pass two lists to this method.

Reply:
Doing the two lists works well.

I'm going to investigate who needs the analyzers. Its weird not to have any analyzers if the field type you are using has some. But it is kind of also weird to copy the ones from the search analyzer.

Reply:
I think I've pushed a working version that isolates the analyzer bit to just the test that needs it.